components/service/experiments/src/main/java/mozilla/components/service/experiments/ExperimentsUpdater.kt

```diff
@@ -84,3 +123,23 @@ internal class ExperimentsUpdater(
         private const val EXPERIMENTS_COLLECTION_NAME = "focus-experiments"
     }
 }
+
+/**
+ * This class represents a [Worker] object that periodically updates the experiments.
+ * @param updater used so that the updater can be injected from tests.
+ */
+internal class ExperimentsUpdaterWorker(
+    context: Context,
+    workerParameters: WorkerParameters,
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal var updater: ExperimentsUpdater = Experiments.updater
+) : Worker(context, workerParameters) {
+
+    override fun doWork(): Result {
+        return if (updater.updateExperiments()) {
+            Result.success()
+        } else {
+            Result.failure()
```

Comment:
What happens on failure?
- Do we get automatic retry from work manager?
- Or do we need a follow-up bug for retry behavior?

Reply:
No, this is an oversight on my part.  I _intended_ to return `Result.retry()` here.

Reply:
Actually, what I think I was thinking is that `Result.failure()` would just give up this attempt and it would try again at the next periodic scheduled interval.

Reply:
I was planning on setting up a manual test today using samples-glean to run the update on the minimum scheduled interval (15 minutes :( ) in order to make sure the periodic part was working and to measure the drift.  I'll try it with both `.failure()` and `.retry()` in order to see what the difference is, but I suspect that `retry` will retry immediately and `failure` will run at the next scheduled interval.

Reply:
We should retry on failure and probably quickly initially.
Maybe we can find out what the normandy behavior is and align with that?

Reply:
Normandy client behavior is reported to be to fail and retry in 6 hours, with the possibility that this may change in the future.  Since this is easy to iterate on, I'm going to verify that returning `Result.failure()` does what we expect (i.e. matches current Normandy client behavior) and call this good for now.